docs/en/sql-reference/dictionaries/external-dictionaries/external-dicts-dict-layout.md

```diff
@@ -156,6 +156,33 @@ or
 LAYOUT(HASHED(PREALLOCATE 0))
 ```
 
+If `shards` greater then 1 (default is `1`) the dictionary will load data in parallel, useful if you have huge amount of elements in one dictionary.
+
+Configuration example:
+
+``` xml
+<layout>
+  <hashed>
+    <shards>10</shards>
+    <!-- Size of the backlog for blocks in parallel queue.
+
+         Since the bottleneck in parallel loading is rehash, and so to avoid
+         stalling because of thread is doing rehash, you need to have some
+         backlog.
+
+         10000 is good balance between memory and speed.
+         Even for 10e10 elements and can handle all the load without starvation. -->
+    <per_shard_load_backlog>10000</per_shard_load_backlog>
+  </hashed>
+</layout>
+```
+
+or
+
+``` sql
+LAYOUT(HASHED(SHARDS 10 [PER_SHARD_LOAD_BACKLOG 10000]))
```

Comment:
Also I am not sure if this needs to be implemented only for hashed dictionary. Maybe it better to make some sort of ShardedDictionary that will wrap any dictionary. That way for client it will be easier to use this thing.

Reply:
It seems that this is pretty specific use case, and I would rather avoid having extra code for other dictionaries rights now.